cli/exchange/service.go

```diff
@@ -139,6 +139,15 @@ func ServicePublish(org, userPw, jsonFilePath, keyFilePath, pubKeyFilePath strin
 	if svcFile.Org != "" && svcFile.Org != org {
 		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("the org specified in the input file (%s) must match the org specified on the command line (%s)", svcFile.Org, org))
 	}
+	if public != "" {
+		public = strings.ToLower(public)
+		if public != "true" && public != "false" {
+			cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("Need to set 'true' or 'false' when specifying flag -V."))
+		} else {
+			// Override public key with key set in the hzn command
+			svcFile.Public, _ = strconv.ParseBool(public)
```

Comment:
I didnt notice this last review. If you ignore the error on the string convert, and there is an error, the pubic field will get set to false. I think you need to check for an error here and report it as fatal.